Weird... It is running it:

> Collecting django-cors-headers (from -r requirements.txt (line 5))
>   Downloading django_cors_headers-2.0.2-py2.py3-none-any.whlCollecting django-cors-headers (from -r requirements.txt (line 5))
>   Downloading django_cors_headers-2.0.2-py2.py3-none-any.whl

But then:


> ERROR: django_example.tests (unittest.loader._FailedTest)
> ----------------------------------------------------------------------
> ImportError: Failed to import test module: django_example.tests
> Traceback (most recent call last):
>   File "/opt/python/3.6.1/lib/python3.6/unittest/loader.py", line 428, in _find_test_path
>     module = self._get_module_from_name(name)
>   File "/opt/python/3.6.1/lib/python3.6/unittest/loader.py", line 369, in _get_module_from_name
>     __import__(name)
>   File "/home/travis/build/millerf/django-channels-jsonrpc/example/django_example/tests.py", line 2, in <module>
>     from channels_jsonrpc import JsonRpcConsumerTest, JsonRpcException
>   File "/home/travis/build/millerf/django-channels-jsonrpc/.tox/coverage/lib/python3.6/site-packages/channels_jsonrpc/__init__.py", line 1, in <module>
>     from .jsonrpcconsumer import JsonRpcConsumer, JsonRpcConsumerTest, JsonRpcException
>   File "/home/travis/build/millerf/django-channels-jsonrpc/.tox/coverage/lib/python3.6/site-packages/channels_jsonrpc/jsonrpcconsumer.py", line 19, in <module>
>     from corsheaders.middleware import CorsMiddleware
> ModuleNotFoundError: No module named 'corsheaders'ERROR: django_example.tests (unittest.loader._FailedTest)
> ----------------------------------------------------------------------
> ImportError: Failed to import test module: django_example.tests
> Traceback (most recent call last):
>   File "/opt/python/3.6.1/lib/python3.6/unittest/loader.py", line 428, in _find_test_path
>     module = self._get_module_from_name(name)
>   File "/opt/python/3.6.1/lib/python3.6/unittest/loader.py", line 369, in _get_module_from_name
>     __import__(name)
>   File "/home/travis/build/millerf/django-channels-jsonrpc/example/django_example/tests.py", line 2, in <module>
>     from channels_jsonrpc import JsonRpcConsumerTest, JsonRpcException
>   File "/home/travis/build/millerf/django-channels-jsonrpc/.tox/coverage/lib/python3.6/site-packages/channels_jsonrpc/__init__.py", line 1, in <module>
>     from .jsonrpcconsumer import JsonRpcConsumer, JsonRpcConsumerTest, JsonRpcException
>   File "/home/travis/build/millerf/django-channels-jsonrpc/.tox/coverage/lib/python3.6/site-packages/channels_jsonrpc/jsonrpcconsumer.py", line 19, in <module>
>     from corsheaders.middleware import CorsMiddleware
> ModuleNotFoundError: No module named 'corsheaders'



That happens on all python versions as well... 

See (not sure if you have access)
https://travis-ci.org/millerf/django-channels-jsonrpc/jobs/234817698

--- a/setup.py
+++ b/setup.py
@@ -12,7 +12,7 @@ setup(
     version='1.1.9',
     packages=find_packages(),
     install_requires=[
-          'channels',
+          'channels', 'django-cors-headers'
       ],
     include_package_data=True,
     license='MIT License',
@@ -22,7 +22,7 @@ setup(
     author='Fabien Millerand - MILLER/f',
     author_email='fab@millerf.com',
     test_suite='channels_jsonrpc.tests.tests',
-    tests_require=['django', 'channels'],
+    tests_require=['django', 'channels', 'django-cors-headers'],
     classifiers=[
         'Environment :: Web Environment',
         'Framework :: Django',

